storage/entities/peer.js

```diff
@@ -0,0 +1,307 @@
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const ft = require('../utils/filter_types');
+const { stringToByte } = require('../utils/inputSerializers');
+const { NonSupportedFilterTypeError } = require('../errors');
+const BaseEntity = require('./base_entity');
+
+const defaultCreateValues = {};
+
+const readOnlyFields = [];
+
+/**
+ * Basic Peer
+ * @typedef {Object} BasicPeer
+ * @property {number} id
+ * @property {inet} ip
+ * @property {number} wsPort
+ * @property {number} state
+ * @property {string} os
+ * @property {string} version
+ * @property {number} clock
+ * @property {string} broadhash
+ * @property {number} height
+ */
+
+/**
+ * Peer Filters
+ * @typedef {Object} filters.Peer
+ */
+
+/**
+ * @typedef {string} fieldSets.Peer
+ * @enum
+ * @value 'FIELD_SET_SIMPLE'
+ */
+
+const FIELD_SET_SIMPLE = Symbol('FIELD_SET_SIMPLE');
+
+class Peer extends BaseEntity {
+	/**
+	 * Constructor
+	 * @param {BaseAdapter} adapter - Adapter to retrieve the data from
+	 * @param {filters.Peer} defaultFilters - Set of default filters applied on every query
+	 */
+	constructor(adapter, defaultFilters = {}) {
+		super(adapter, defaultFilters);
+
+		this.defaultFilters = defaultFilters;
+		this.overrideDefaultOptions({ fieldSet: FIELD_SET_SIMPLE });
+
+		this.addField('id', 'number', { filter: ft.NUMBER });
+		this.addField('ip', 'string', { format: 'ip', filter: ft.TEXT });
+		this.addField('wsPort', 'number', {
+			format: 'number',
+			filter: ft.NUMBER,
+		});
+		this.addField('state', 'number', { format: 'number', filter: ft.NUMBER });
+		this.addField('os', 'string', { format: 'string', filter: ft.TEXT });
+		this.addField('version', 'string', { format: 'string', filter: ft.TEXT });
+		this.addField('clock', 'number', { format: 'number', filter: ft.NUMBER });
+		this.addField(
+			'broadhash',
+			'string',
+			{ format: 'string', filter: ft.TEXT },
+			stringToByte
+		);
+		this.addField('height', 'number', {
+			format: 'number',
+			filter: ft.NUMBER,
+		});
+
+		this.SQLs = {
+			selectSimple: this.adapter.loadSQLFile('peers/get_simple.sql'),
+			create: this.adapter.loadSQLFile('peers/create.sql'),
+			update: this.adapter.loadSQLFile('peers/update.sql'),
+			updateOne: this.adapter.loadSQLFile('peers/update_one.sql'),
+			isPersisted: this.adapter.loadSQLFile('peers/is_persisted.sql'),
+		};
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	get FIELD_SET_SIMPLE() {
+		return FIELD_SET_SIMPLE;
+	}
+
+	/**
+	 * Get one peer
+	 *
+	 * @param {filters.Peer|filters.Peer[]} [filters = {}]
+	 * @param {Object} [options = {}] - Options to filter data
+	 * @param {Number} [options.limit=10] - Number of records to fetch
+	 * @param {Number} [options.offset=0] - Offset to start the records
+	 * @param {fieldSets.Peer} [options.fieldSet='FIELD_SET_SIMPLE'] - Fieldset to choose
```

Comment:
Didn't we agree to use `options.extended` instead of keeping the fieldSets as constant?

Reply:
We agreed to change that in a separated issue